parlai/agents/alice/alice.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Agent uses the Free Alice AIML interpreter to generate replies to observations.
+"""
+
+import os
+from parlai.core.agents import Agent
+
+try:
+    import aiml
+except ImportError:
+    raise ImportError(
+        (
+            "ALICE agent needs python-aiml installed. Please run:\n "
+            "pip install git+https://github.com/paulovn/python-aiml.git"
+        )
+    )
+
+
+class AliceAgent(Agent):
+    """
+    Agent returns the Alice AIML bot's reply to an observation.
+
+    This is a strong rule-based baseline.
+    """
+
+    def __init__(self, opt, shared=None):
+        """
+        Initialize this agent.
+        """
+        super().__init__(opt)
+        self.id = 'Alice'
+        self.kern = None
```

Comment:
let's have `kern` be shared

so it will be like this:

```python
if shared is None:
    self._load_alice()
else:
    self.kern = shared['kern']
```

and then below
```python
def share(self):
    shared = super().share()
    shared['kern'] = self.kern
    return shared
```